File: swift/nekomimi.swift
var input = readLine()!
let cat = "cat"

var foundCnt = 0

for idx in 0 ... input.characters.count - cat.characters.count {
    var idxInput = input.startIndex.advancedBy(idx)
    var idxCat = cat.startIndex;
    if (input[idxInput] == cat[idxCat]) {
        var found = true
        for subIdx in 1 ..< cat.characters.count {
            if (input[idxInput.successor()] != cat[idxCat.successor()]) {
                found = false
                break
            }
        }
        if (found) {
            foundCnt += 1
        }
    }
}

print(foundCnt)

File: cpp/nekomimi.cpp
#include <iostream>
using namespace std;

string readline() {
    string str;
    getline(cin, str);
    return str;
}

int main(void){
    auto foundCnt = 0;
    const string cat = "cat";
    const auto input = readline();
    for (auto idx = 0; idx <= input.size() - cat.size(); idx++) {
        if (input[idx] == cat[0]) {
            bool found = true;
            for (auto subidx = 1; subidx < cat.size(); subidx++) {
                if (input[idx + subidx] != cat[subidx]) {
                    found = false;
                    break;
                }
            }
            if (found) {
                foundCnt++;
            }
        }
    }
    cout << foundCnt << endl;
    return 0;
}

